src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -956,21 +961,29 @@ public boolean isSneaking()
 	@Override
 	public void setSneaking(boolean sneaking)
 	{
-		this.sneaking = sneaking;
+		PlayerToggleSneakEvent event = new PlayerToggleSneakEvent(this, sneaking);
+		Bukkit.getPluginManager().callEvent(event);
+		if (!event.isCancelled())
+		{
+			this.sneaking = event.isSneaking();
+		}
 	}
 
 	@Override
 	public boolean isSprinting()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return sprinting;
 	}
 
 	@Override
 	public void setSprinting(boolean sprinting)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		PlayerToggleSprintEvent event = new PlayerToggleSprintEvent(this, sprinting);
+		Bukkit.getPluginManager().callEvent(event);
+		if (!event.isCancelled())
+		{
+			this.sprinting = event.isSprinting();
+		}
```

Comment:
This is an erroneous implementation.
CraftBukkit does not call such an event when invoking this method.
Setters only modify the state whereas any PlayerEvent is the result of the actions from an actual Player, you would need a simulation method for this. The setter itself doesn't call an event.